test/MarginAccount.test.ts

```diff
@@ -0,0 +1,485 @@
+import {MarginAccountTesterInstance, MockERC20Instance, OtokenInstance} from '../build/types/truffle-types'
+
+import {BigNumber} from 'bignumber.js'
+
+const {expectRevert} = require('@openzeppelin/test-helpers')
+
+const Otoken = artifacts.require('Otoken.sol')
+const MockERC20 = artifacts.require('MockERC20.sol')
+const MarginAccountTester = artifacts.require('MarginAccountTester.sol')
+const MockAddressBook = artifacts.require('MockAddressBook')
+const ZERO_ADDR = '0x0000000000000000000000000000000000000000'
+const ETH_ADDR = ZERO_ADDR
+
+contract('MarginAccount', ([deployer, controller]) => {
+  // ERC20 mocks
+  let weth: MockERC20Instance
+  // addressbook instance
+  let marginAccountTester: MarginAccountTesterInstance
+  let otoken: OtokenInstance
+  let otoken2: OtokenInstance
+  let usdc: MockERC20Instance
+  let mockAddressBookAddr: string
+
+  // let expiry: number;
+  const strikePrice = new BigNumber(200).times(new BigNumber(10).exponentiatedBy(18))
+  const expiry = 1601020800 // 2020/09/25 0800 UTC
+  const isPut = true
+
+  const testAccount: {owner: string; vaultIds: BigNumber} = {owner: ZERO_ADDR, vaultIds: new BigNumber(0)}
+  let account: {owner: string; vaultIds: BigNumber}
+
+  before('Deployment', async () => {
+    // deploy WETH token
+    weth = await MockERC20.new('WETH', 'WETH')
+    // deploy AddressBook token
+    marginAccountTester = await MarginAccountTester.new()
+
+    const addressBook = await MockAddressBook.new()
+    mockAddressBookAddr = addressBook.address
+    await addressBook.setController(controller)
+
+    // deploy oToken with addressbook
+    otoken = await Otoken.new(addressBook.address)
+
+    usdc = await MockERC20.new('USDC', 'USDC')
+
+    // deploy second otoken
+    otoken2 = await Otoken.new(addressBook.address)
+    await otoken2.init(ETH_ADDR, usdc.address, usdc.address, strikePrice, expiry, isPut, {from: deployer})
+  })
+
+  describe('Otoken Initialization', () => {
+    it('should be able to initialize with put parameter correctly', async () => {
+      await otoken.init(ETH_ADDR, usdc.address, usdc.address, strikePrice, expiry, isPut, {from: deployer})
+      assert.equal(await otoken.underlyingAsset(), ETH_ADDR)
+      assert.equal(await otoken.strikeAsset(), usdc.address)
+      assert.equal(await otoken.collateralAsset(), usdc.address)
+      assert.equal((await otoken.strikePrice()).toString(), strikePrice.toString())
+      assert.equal(await otoken.isPut(), isPut)
+      assert.equal((await otoken.expiry()).toNumber(), expiry)
+    })
```

Comment:
1. Remove the `describe("Otoken Initialization" ()=> {}` block
2. Remove all Otoken related setup / tests. Just use MockERC20 for your `otoken` as in this PR:
https://github.com/opynfinance/GammaProtocol/pull/51/files